### PR TITLE
APIGOV-22682 - update for access request provisioning data

### PR DESCRIPTION
**Pull request by jcollins-axway**

### CODE REVIEW

**Review by sbolosan — APPROVED** [review]

**Review by vivekschauhan — APPROVED** [review]

File: pkg/agent/discoverycache.go [hunk 1]
```diff
@@ -126,7 +126,7 @@ func (j *discoveryCache) updateAPIServiceInstancesCache() {
 	defer j.instanceCacheLock.Unlock()
 
 	query := map[string]string{
-		apic.FieldsKey: apiServerFields,
+		apic.FieldsKey: apiServerFields + "," + defs.Spec,
```
Comment (tsjohns9) [inline pkg/agent/discoverycache.go:129]: I removed queries in the discovery cache

File: pkg/agent/handler/accessrequest.go [hunk 2]
```diff
@@ -20,22 +20,24 @@ const (
...
 // NewAccessRequestHandler creates a Handler for Access Requests
 func NewAccessRequestHandler(prov arProvisioner, cache agentcache.Manager, client client) Handler {
 	return &accessRequestHandler{
-		prov:   prov,
-		cache:  cache,
-		client: client,
+		prov:          prov,
+		cache:         cache,
+		client:        client,
+		encryptSchema: encryptSchema,
```
Comment (tsjohns9) [inline pkg/agent/handler/accessrequest.go:40]: Where does encryptSchema come from?

File: pkg/agent/handler/accessrequest.go [hunk 3]
```diff
@@ -56,6 +58,11 @@ func (h *accessRequestHandler) Handle(ctx context.Context, meta *proto.EventMeta
 		return nil
 	}
 
+	// add or update the cache with the access request
+	if action == proto.Event_CREATED || action == proto.Event_UPDATED {
```
Comment (tsjohns9) [inline pkg/agent/handler/accessrequest.go:62]: How come the resource gets cached before processing?

File: pkg/agent/handler/accessrequest.go (hunk 2 above)
Comment (jcollins-axway) [reply pkg/agent/handler/accessrequest.go:40]: Its the same that the credential uses

File: pkg/agent/handler/accessrequest.go (hunk 3 above)
Comment (jcollins-axway) [reply pkg/agent/handler/accessrequest.go:62]: Want to make sure our cache has the latests, on an update event we may not have to process anything but want the cache updated

**Review by tsjohns9 — APPROVED** [review]

**Review by vivekschauhan — APPROVED** [review]

**Pull request merged**